main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -142,27 +142,23 @@ object Lowering {
     * Translates internal Datalog constraints into Flix Datalog constraints.
     */
   def run(root: TypedAst.Root)(implicit flix: Flix): LoweredAst.Root = flix.phase("Lowering") {
-    val defs = ParOps.parMap(root.defs.values)((d: TypedAst.Def) => visitDef(d)(root, flix))
-    val sigs = ParOps.parMap(root.sigs.values)((s: TypedAst.Sig) => visitSig(s)(root, flix))
-    val instances = ParOps.parMap(root.instances.values)((insts: List[TypedAst.Instance]) => insts.map(i => visitInstance(i)(root, flix)))
-    val enums = ParOps.parMap(root.enums.values)((e: TypedAst.Enum) => visitEnum(e)(root, flix))
-    val restrictableEnums = ParOps.parMap(root.restrictableEnums.values)((e: TypedAst.RestrictableEnum) => visitRestrictableEnum(e)(root, flix))
-    val effects = ParOps.parMap(root.effects.values)((e: TypedAst.Effect) => visitEffect(e)(root, flix))
-    val aliases = ParOps.parMap(root.typeAliases.values)((a: TypedAst.TypeAlias) => visitTypeAlias(a)(root, flix))
-
-    val newDefs = defs.map(kv => kv.sym -> kv).toMap
-    val newSigs = sigs.map(kv => kv.sym -> kv).toMap
-    val newInstances = instances.map(kv => kv.head.clazz.sym -> kv).toMap
-    val newEnums = (enums ++ restrictableEnums).map(kv => kv.sym -> kv).toMap
-    val newEffects = effects.map(kv => kv.sym -> kv).toMap
-    val newAliases = aliases.map(kv => kv.sym -> kv).toMap
+    val defs = ParOps.mapValues(root.defs)((d: TypedAst.Def) => visitDef(d)(root, flix))
```

Comment:
Is it possible to simply write `ParOps.mapValues(root.defs)(visitDef)` or is Scala's type inference too weak?